src/gateways/notes/NotesResponseMapper.ts

```diff
@@ -0,0 +1,31 @@
+import { INote, NoteType } from "entities/INote";
+import { IEntityLink } from "../../entities/IEntityLink";
+import { isoDate } from "../../utils/dateTime";
+import { INotesResponse } from "./INotesResponse";
+
+export interface INotesResponseMapper {
+  map: (notesApiResponse: INotesResponse) => INote[];
+}
+
+export class NotesResponseMapper implements INotesResponseMapper {
+  public map(notesApiResponse: INotesResponse): INote[] {
+    return notesApiResponse.data.map((note) => {
```

Comment:
Could this method be named `toINotes` or similar to avoid the overloading of the `.map` method name?  

I see `.map` and I think "ah this is a method that applies a callback to each element in an array (like on line 12), but wait `INotesResponse` isn't an array!" what's going on?!

Reply:
but INotesResponse **is** an array 😋

Reply:
INotesResponse is an array called data in a jsonapi wrapper, this map function does (from a black box point of view) apply a transformation over each element of that data array and returns a new INote for each.